test/logger_acc.erl

```diff
@@ -0,0 +1,98 @@
+%% @doc A logger report handler that be used to capture expected errors in
+%% tests. The current error report handlers are disabled during the execution
+%% of a function. Afterwards, they are restored and the errors that occered are
+%% returned along with the return value of the fun.
+-module(logger_acc).
+
+%% Public API
+-export([capture/1]).
+
+%% @doc Executes `Fun' and captures all logged errors returns as well as
+%% uncaught errors in `Fun'.
+-spec capture(fun (() -> ResultOfFun)) ->
+    {ok, ResultOfFun, AccumulatedErrors} |
+    {throw | error | exit, Reason, Trace, AccumulatedErrors}
+  when ResultOfFun :: term(),
+       Reason :: term(),
+       Trace :: list(),
+       AccumulatedErrors :: [{error|warning_msg|info_msg, string()} |
+                             {error_report|warning_report|info_report, term()}].
+capture(Fun) when is_function(Fun, 0) ->
+    Tag = make_ref(),
+    Self = self(),
+    AccPid = spawn_link(fun() -> log_acc_loop(Tag, Self) end),
+    logger:add_primary_filter(?MODULE, {fun(Event, _)-> AccPid ! {Tag, Event}, stop end, undefined}),
+    try
+        Fun()
+    of
+        Result ->
+            Events = flush_logs(AccPid, Tag),
+            {ok, Result, Events}
+    catch
+        Class:Error:Stacktrace ->
+            Events = flush_logs(AccPid, Tag),
+            {Class, Error, Stacktrace, Events}
+    after
+        ok = logger:remove_primary_filter(?MODULE)
+    end.
+
+flush_logs(AccPid, Tag) ->
+    AccMon = monitor(process, AccPid),
+    AccPid ! {Tag, flush},
+    receive
+        {Tag, Events} ->
+            demonitor(AccMon, [flush]),
+            Events;
+        {'DOWN', AccMon, process, AccPid, Reason} ->
+            error({accumulator_process, Reason})
+    end.
```

Comment:
Do we need a timeout for this receive? Or not needed when we have the monitor?

Reply:
No need I think, from how the accumulating loop is designed.

Reply:
OK, and it's just a testing helper, so worst case the test will time out.